numpy/core/src/multiarray/buffer.c

```diff
@@ -706,7 +771,9 @@ array_getbuffer(PyObject *obj, Py_buffer *view, int flags)
     }
 
     /* Fill in information */
-    info = _buffer_get_info(obj);
+    info = _buffer_get_info(obj,
+            (flags & PyBUF_F_CONTIGUOUS) == PyBUF_F_CONTIGUOUS,
+            (flags & PyBUF_FORMAT) == PyBUF_FORMAT);
```

Comment:
it's looking increasingly like maybe we should just pass in the full flags argument

Reply:
Hmmm, fair point, could pass in the flags.

Reply:
Is this still relevant?

Reply:
Changed it and rebased. The rebase means the code changed slightly here (basically, I undid a bit of fixup that I did in the other PR to make this one easier, because I got it wrong...).